src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -361,21 +361,25 @@ def sync_deploy_root_with_stage(
 
         files_not_removed = []
         if local_paths_to_sync:
-            assert mapped_files is not None
+            assert bundle_map is not None
 
             # Deploying specific files/directories
             resolved_paths_to_sync = [
                 resolve_without_follow(p) for p in local_paths_to_sync
             ]
             if not recursive:
                 verify_no_directories(resolved_paths_to_sync)
-            deploy_paths_to_sync = [
-                source_path_to_deploy_path(p, mapped_files)
-                for p in resolved_paths_to_sync
-            ]
-            verify_exists(deploy_paths_to_sync)
+
+            deploy_paths_to_sync = []
+            for resolved_path in resolved_paths_to_sync:
+                verify_exists(resolved_path)
+                deploy_path = bundle_map.to_deploy_path(resolved_path)
+                if deploy_path is None:
+                    raise ClickException(f"No artifact found for {resolved_path}")
+                deploy_paths_to_sync.append(deploy_path)
+
             stage_paths_to_sync = _get_stage_paths_to_sync(
-                deploy_paths_to_sync, self.deploy_root.resolve()
+                deploy_paths_to_sync, resolve_without_follow(self.deploy_root)
```

Comment:
Is this a bugfix?

Reply:
Kind of, it ends up making things inconsistent on Windows because the specific resolution logic you use matters when doing prefix comparison. We use resolve_without_follow everywhere except for here, and without this change some integ tests fail due to the weird ~1 expansion windows does sometimes.